staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -133,13 +133,23 @@ func CheckDiffErr(err error) {
 	})
 }
 
+// unwrapStatusError returns the unwrapped StatusError and the flag indicates if
+// it is a StatusError / wrapped StatusError or not.
+func unwrapStatusError(err error) (*apierrors.StatusError, bool) {
```

Comment:
Why do we need this helper that manually recreates the error and falls back on checking the top-level error? Under what circumstances can L142 be hit? 

I would expect we'd be able to simply replace the type assertion with `errors.As`:

```diff
diff --git a/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go b/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go
index caebb46c85e..f243e36def1 100644
--- a/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go
@@ -149,8 +149,8 @@ func isInvalidReasonStatusError(err error) bool {
        if !apierrors.IsInvalid(err) {
                return false
        }
-       statusError, isStatusError := unwrapStatusError(err)
-       if !isStatusError {
+       var statusError *apierrors.StatusError
+       if !errors.As(err, &statusError) {
                return false
        }
        status := statusError.Status()
@@ -173,7 +173,8 @@ func checkErr(err error, handleErr func(string, int)) {
        case err == ErrExit:
                handleErr("", DefaultErrorExitCode)
        case isInvalidReasonStatusError(err):
-               statusError, _ := unwrapStatusError(err)
+               var statusError *apierrors.StatusError
+               _ = errors.As(err, &statusError)
                status := statusError.Status()
                details := status.Details
                s := "The request is invalid"
```

Reply:
Great suggestion. I made changes as you suggested. I was just thinking if the errors.As is not correct in some unexpected cases (though I do not find any by far), using statusError.Status() would cause panic so I wrote some guard logic to that.